python/tests/kat/t_tcpmapping.py

```diff
@@ -264,3 +265,868 @@ def check(self):
+  port: 6789
+  service: {self.target.path.fqdn}:80
+''') + super().manifests()
+
+    def queries(self):
+        yield Query(self.url("", port=6789))
+
+    def check(self):
+        assert self.results[0].json["backend"] == self.target.path.k8s
+        assert self.results[0].json["request"]["tls"]["enabled"] == False
+
+
+class TCPMappingTLSOriginationBoolTest(AmbassadorTest):
+    extra_ports = [ 6789 ]
+    target: ServiceType
+
+    def init(self) -> None:
+        self.target = HTTP()
+
+    def manifests(self) -> str:
+        return format('''
+---
+apiVersion: getambassador.io/v2
+kind: TCPMapping
+metadata:
+  name: {self.name.k8s}
+spec:
+  ambassador_id: [ {self.ambassador_id} ]
+  port: 6789
+  service: {self.target.path.fqdn}:443
+  tls: true
+''') + super().manifests()
+
+    def queries(self):
+        yield Query(self.url("", port=6789))
+
+    def check(self):
+        assert self.results[0].json["backend"] == self.target.path.k8s
+        assert self.results[0].json["request"]["tls"]["enabled"] == True
+
+
+class TCPMappingTLSOriginationV2SchemeTest(AmbassadorTest):
+    """apiVersion v2 TCPMappings don't support a scheme:// on the 'service' field; if you provide
+    one, then it is ignored.  Since apiVersion v3alpha1 adds support for scheme://, add a test to
+    make sure we don't break anyone who is inadvertently depending on it being ignored in v2."""
+
+    extra_ports = [ 6789, 6790 ]
+    target: ServiceType
+
+    def init(self) -> None:
+        self.xfail = "bug (2.3): v2 TCPMappings don't ignore the scheme"
+        self.target = HTTP()
+
+    def manifests(self) -> str:
+        return format('''
+---
+apiVersion: getambassador.io/v2
+kind: TCPMapping
+metadata:
+  name: {self.name.k8s}-1
+spec:
+  ambassador_id: [ {self.ambassador_id} ]
+  port: 6789
+  service: https://{self.target.path.fqdn}:443
+---
+apiVersion: getambassador.io/v2
+kind: TCPMapping
+metadata:
+  name: {self.name.k8s}-2
+spec:
+  ambassador_id: [ {self.ambassador_id} ]
+  port: 6790
+  service: https://{self.target.path.fqdn}:80
+''') + super().manifests()
+
+    def queries(self):
+        yield Query(self.url("", port=6789), expected=400) # kat-server returns HTTP 400 "Client sent an HTTP request to an HTTPS server."
+        yield Query(self.url("", port=6789, scheme='https'), insecure=True)
+        yield Query(self.url("", port=6790))
+
+    def check(self):
+        assert self.results[1].json["backend"] == self.target.path.k8s
+        assert self.results[1].json["request"]["tls"]["enabled"] == True
+        assert self.results[2].json["backend"] == self.target.path.k8s
+        assert self.results[2].json["request"]["tls"]["enabled"] == False
+
+
+class TCPMappingTLSOriginationV3SchemeTest(AmbassadorTest):
+    extra_ports = [ 6789, 6790 ]
+    target: ServiceType
+
+    def init(self) -> None:
+        self.target = HTTP()
+
+    def manifests(self) -> str:
+        return format('''
+---
+apiVersion: getambassador.io/v3alpha1
```

Comment:
Oops, it was a mistake that I left `6790` in `extra_ports`.

The difference is that in v2 it didn't support saying "originate TLS by adding `https://` to the `service`; but in v3alpha1 this is the **primary** way of originating TLS without a client cert (since v3alpha1 doesn't support `tls: true`, which is how you did that in v2.)

Reply:
I've added a commit removing `6790` from `extra_ports`.